### ISSUE TITLE
Wrong terminal in dev containers

### ISSUE THREAD

**Issue by sgraband** [eclipse-theia/theia#14293]
### Bug Description:
When opening a development container, the terminal that is opened defaults to a simple `sh` shell rather than the user's configured default terminal (such as `bash`, `zsh`, etc.), which is the case in a normal workspace. This behavior is inconsistent and can cause confusion, as users might expect the same terminal configuration in both normal workspaces and development containers.

### Steps to Reproduce:
1. Open a normal workspace and observe that the terminal uses the user's default shell (e.g., `bash`, `zsh`).
2. Open the same workspace as a development container.
3. Observe that the terminal defaults to `sh` instead of the user's default terminal.

### Expected Behavior:
The terminal in a development container should respect the user's default shell, just as it does in a normal workspace.

### Additional Information:
- Operating System: Ubuntu 22.04
- Theia Version: v1.54.0

**Comment by jonah-iden** [comment]
This should probably be fixed through https://github.com/eclipse-theia/theia/pull/14548

**Comment by jonah-iden** [comment]
@sgraband To continue the discussion from the PR here:
So what you were saying is, when connected to a dev container the default terminal should always be bash (execpt of course if something else is configured). 
You are on Linux right? Is theias default terminal on linux normally bash or did you have to configure that?
If you had to configure it then maybe it would rather be better to in general use bash as default if installed. Maybe add a small check somewhere in the terminal package

**Comment by sgraband** [comment]
Yes i am on Linux. Theias default Terminal is afaik (i haven't found any settings i made in this regard) the default shell of the users system (in my case zsh). This would of course be the ideal solution, but this is probably hard to do in the dev-container i assume.

(I have `"terminal.integrated.profiles.linux": null` locally as well on the dev-container.)

The thing is that the default shell in the dev-container is basically unusable in most of the devcontainers i am using, switching by hand to `bash` always improves the situation (and is the behaviour expected by users migrating from VSCode).

I'm not sure if this comes from the way the remote is setup or if we somehow set the terminal to a `shell` because we could not load the users configuration.

**Comment by jonah-iden** [comment]
yeah if the default terminal is configured outside of theia we can't really reach that from inside the dev-container. I'll check tomorrow how the terminal package determines that. Maybe 
Thanks for the explanation

**Comment by JonasHelming** [comment]
Is there any update on this?

**Comment by jonah-iden** [comment]
Sadly not. I havn't found the time to have another look at this yet

**Comment by eneufeld** [comment]
I would find it much more necessary to have have my system environment variables in the container. When starting the container I can't use Theia AI as none of my environment variables are available.
Can we do this as part of this issue or should I open a new one?

**Comment by jonah-iden** [comment]
@eneufeld it would probably be the best to open a new Issue for that since this issue is more about settings. Env variables would be a different feature
